rest/route/project_validate_test.go

```diff
@@ -0,0 +1,60 @@
+package route
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"net/http"
+	"testing"
+
+	"github.com/evergreen-ci/evergreen/db"
+	"github.com/evergreen-ci/evergreen/model"
+	"github.com/evergreen-ci/evergreen/validator"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestValidateProjectHandlerValid(t *testing.T) {
+	invalidYml := `
+tasks:
+  - name: my_task
+
+buildvariants:
+  - name: my_build_variant
+    display_name: My Build Variant
+    run_on:
+    - "not_real"
+    tasks:
+    - name: my_task
+`
+
+	require.NoError(t, db.ClearCollections(model.ProjectRefCollection))
+	ref := model.ProjectRef{Id: "proj"}
+	require.NoError(t, ref.Insert())
+
+	input := validator.ValidationInput{
+		ProjectYaml: []byte(invalidYml),
+		ProjectID:   "proj",
+	}
+	bodyBytes, err := json.Marshal(input)
+	require.NoError(t, err)
+
+	req, err := http.NewRequest(http.MethodPost, "/validate", bytes.NewBuffer(bodyBytes))
+	require.NoError(t, err)
+
+	handler := makeValidateProject().Factory()
+	require.NoError(t, handler.Parse(context.Background(), req))
```

Comment:
Instead of `context.Background()`, can you use `t.Context()`?

Reply:
good idea, yes! 🎉